src/cdk-experimental/testing/harness-environment.ts

```diff
@@ -7,14 +7,34 @@
  */
 
 import {
-  AsyncFn,
+  AsyncFactoryFn,
   ComponentHarness,
   ComponentHarnessConstructor,
   HarnessLoader,
+  HarnessPredicate,
   LocatorFactory
 } from './component-harness';
 import {TestElement} from './test-element';
 
+function _getErrorForMissingSelector(selector: string): Error {
+  return Error(`Expected to find element matching selector: "${selector}", but none was found`);
+}
+
+function _getErrorForMissingHarness<T extends ComponentHarness>(
+    harnessType: ComponentHarnessConstructor<T> | HarnessPredicate<T>): Error {
+  const harnessPredicate =
+      harnessType instanceof HarnessPredicate ? harnessType : new HarnessPredicate(harnessType);
+  const name = harnessPredicate.harnessType.name;
+  const selector = harnessPredicate.harnessType.hostSelector;
```

Comment:
Looks like you can destructure these two variables: `{name, hostSelector: selector} = harnessPredicate.harnessType;`